src/ClientOptionsTrait.php

```diff
@@ -80,8 +81,20 @@ private static function getClientDefaults()
         return [];
     }
 
-    private function buildClientOptions(array $options)
+    private function buildClientOptions(array|ClientOptions $options)
     {
+        if ($options instanceof ClientOptions) {
+            $options = $options->toArray();
+
+            // The constructor of ClientOptions construct creates a `TransportOptions`.
+            // We call `toArray` on it and nested items that uses the `optionsTrait` to keep
+            // the same behaviour as if it was just an array.
```

Comment:
We could also make `toArray` call `toArray` on those items.... it would be a breaking change, so I think it's a bad idea. It's a shame we didn't do that before! Maybe in a 2.0 version...